include/odp/api/spec/ipsec.h

```diff
@@ -1293,6 +1297,10 @@ int odp_ipsec_in(const odp_packet_t pkt_in[], int num_in,
  * with IPSEC, etc headers constructed according to the standards. The amount
  * and content of packet data before the IP header is undefined.
  *
+ * Additional TFC padding might be present after packet payload for ESP-tunnel
+ * mode. It will be included into encrypted packet. Receiver side will skip
+ * this padding automatically.
+ *
```

Comment:
What controls the number of such padding bytes? This should be documented.

Reply:
Nothing controls number of padding bytes. It is up to an application to decide.

Reply:
OK, then I would make it clear that these are supplied by the application. For example:
```
The application may choose to insert additional padding bytes after the packet payload (bytes in the packet beyond IP length). These will be included in the encrypted output packet, to be skipped by the receiver.
```
I wouldn't make representations about automatic skipping since  there's no assurance that the receiver is an ODP application so we can't say what some other recipient will do.

Reply:
This is in conflict with this existing statement a little earlier:

"Additionally, input IP packet length (odp_packet_len() minus odp_packet_l3_offset()) must match values in protocol headers. Otherwise esults are undefined."

It could be reworded like this:

"Additionally, input L3 packet length (odp_packet_len() minus odp_packet_l3_offset()) must not be smaller than the IP packet lenght indicated by the IP header. Otherwise results are undefined. If the input L3 packet length is bigger than the IP packet length, the additional packet data is used as TFC padding."

And I think that would be sufficient about TFC here.

Reply:
@JannePeltonen I think @lumag's latest wording is concise. Any application using TFC should be familiar with it's requirements and we shouldn't have to repeat the RFC here.

Reply:
@Bill-Fischofer-Linaro I did not add the reference to the RFC.
@lumag Same comment also here about dropping the remark about the ESP_TFC_PADDING_NOT_SUPPORTED notification. Actually, I think the whole last added paragraph could be just removed.